x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -547,16 +547,16 @@ static Set<RoleDescriptor> maybeRemoveRemoteIndicesPrivileges(
             }).collect(Collectors.toSet());
 
             if (false == affectedRoles.isEmpty()) {
+                final List<String> sortedAffectedRoles = affectedRoles.stream().sorted().toList();
                 logger.info(
-                    "removed remote indices privileges from role(s) [{}] for API key(s) [{}]",
-                    affectedRoles,
+                    "removed remote indices privileges from role(s) {} for API key(s) [{}]",
+                    sortedAffectedRoles,
                     buildDelimitedStringWithLimit(10, apiKeyIds)
                 );
-
                 HeaderWarning.addWarning(
-                    "Removed API key's remote indices privileges from role(s) ["
-                        + affectedRoles
-                        + "]. Remote indices are not supported by all nodes in the cluster. "
```

Comment:
I removed `[]` around `sortedAffectedRoles` since it's already included when `List.toString` gets called.